src/Sarif/Readers/DelegatingStream.cs

```diff
@@ -0,0 +1,51 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.IO;
+
+namespace Microsoft.CodeAnalysis.Sarif
+{
+    internal class DelegatingStream : Stream
+    {
+        private readonly Stream stream;
+
+        /// <summary>
+        /// This is a wrapper for a stream that protects the underlying stream from being disposed.
+        /// Disposing this object is a no-op. To dispose the underlying stream use the DisposeUnderlyingStream method.
+        /// </summary>
+        /// <param name="underlyingStream"></param>
+        internal DelegatingStream(Stream underlyingStream)
```

Comment:
>underlyingStream [](http://example.com/codeflow?start=41&length=16)

what are we supposed to do with a null argument?

Reply:
changed the code to generate a ArgumentNullException